File: app/schemas/checklist.py
"""Pydantic schemas for ChecklistMissao and ItemChecklistMissao."""

from datetime import datetime

from pydantic import BaseModel, Field

from app.models.enums import ChecklistStatusGeral, ItemChecklistStatus


class ItemChecklistMissaoBase(BaseModel):
    nome_item: str = Field(..., max_length=200)
    obrigatorio: bool = True
    status_item: ItemChecklistStatus = ItemChecklistStatus.PENDENTE
    observacao: str | None = None


class ItemChecklistMissaoUpdate(BaseModel):
    status_item: ItemChecklistStatus
    observacao: str | None = None


class ItemChecklistMissaoResponse(ItemChecklistMissaoBase):
    id: int
    checklist_id: int
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class ChecklistMissaoResponse(BaseModel):
    id: int
    missao_id: int
    status_geral: ChecklistStatusGeral
    preenchido_por: int
    revisado_por: int | None
    preenchido_em: datetime
    revisado_em: datetime | None
    observacoes: str | None
    itens: list[ItemChecklistMissaoResponse] = []
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/checklist.ts
/** TypeScript interfaces for Checklist entities matching Pydantic schemas. */

import type { ChecklistStatusGeral, ItemChecklistStatus } from './enums';

export interface ItemChecklistMissaoResponse {
  id: number;
  checklist_id: number;
  nome_item: string;
  obrigatorio: boolean;
  status_item: ItemChecklistStatus;
  observacao: string | null;
  created_at: string;
  updated_at: string;
}

export interface ItemChecklistMissaoUpdate {
  status_item: ItemChecklistStatus;
  observacao?: string | null;
}

export interface ChecklistMissaoResponse {
  id: number;
  missao_id: number;
  status_geral: ChecklistStatusGeral;
  preenchido_por: number;
  revisado_por: number | null;
  preenchido_em: string;
  revisado_em: string | null;
  observacoes: string | null;
  itens: ItemChecklistMissaoResponse[];
  created_at: string;
  updated_at: string;
}
